esmvalcore/_data_finder.py

```diff
@@ -37,15 +37,28 @@ def get_start_end_year(filename):
     stem = Path(filename).stem
     start_year = end_year = None
 
+    # We use (?:) in order not to capture pairs of digits, but
+    # directly whole date
+    date_pattern = r"((?:[0-9]{2}){2,6})"
+    # We use (?:) in order not to captue the range, but directly both
+    # dates separately
+    date_range_pattern = r"(?:" + date_pattern + r"[-_]" + date_pattern + r")"
+    # Next string allows to test that there is an allowed delimiter (or
+    # string start or end) close to date range (or to single date)
+    context = r"(?:^|-|_|$)"
```

Comment:
Why not `[]` syntax? Probably because of `[^]`, right? Might be worth a small comment to prevent future "optimization".

Reply:
Cahnge done. Use [ ] when applicable